DetDescr/src/DetDescr/HcalGeometry.cxx

```diff
@@ -0,0 +1,69 @@
+#include "DetDescr/HcalGeometry.h"
+
+#include <assert.h>
+#include <iostream>
+#include <iomanip>
+
+namespace ldmx {
+
+    HcalGeometry::HcalGeometry(const framework::config::Parameters& ps) : framework::ConditionsObject(HcalGeometry::CONDITIONS_OBJECT_NAME) {
+
+        hcalThicknessScint_ = ps.getParameter<double>("hcalThicknessScint");
+	hcalWidthScint_     = ps.getParameter<double>("hcalWidthScint");
+        hcalZeroLayer_      = ps.getParameter<std::vector< double >>("hcalZeroLayer");
+        hcalZeroStrip_      = ps.getParameter<std::vector< double >>("hcalZeroStrip");
+        hcalLayerThickness_ = ps.getParameter<std::vector< double >>("hcalLayerThickness");
+        hcalNLayers_        = ps.getParameter<std::vector< int >>("hcalNLayers");
+        hcalNStrips_        = ps.getParameter<std::vector< int >>("hcalNStrips");
+        hcalHalfTotalWidthBack_ = ps.getParameter<double>("hcalHalfTotalWidthBack");
+        verbose_            = ps.getParameter<int>("verbose");
+
+	buildStripPositionMap();
+    }
+
+    void HcalGeometry::buildStripPositionMap() {
+
+      for(int section=0; section<=ldmx::HcalID::HcalSection::LEFT; section++) {
```

Comment:
This will break if some future person makes `LEFT` *not* the last section listed. I would suggest, hard-coding the number of sections here and putting in a comment in the definition of `HcalID::HcalSection` enumerator where the number of section is used.

Reply:
I switched this to the `Nsections_` parameter given by `HcalGeometry.py`. Is this in line with what you were thinking? See [here](https://github.com/LDMX-Software/ldmx-sw/blob/51e1bfab6506a789417f7ac3c85197d363f4a58c/DetDescr/src/DetDescr/HcalGeometry.cxx#L28-L30)

Reply:
Yes, I think that makes sense.